File: Medium/MaximumMatchingofPlayersWithTrainers.java
import java.util.Arrays;

public class MaximumMatchingofPlayersWithTrainers {
    public static void main(String[] args) {
        // write code according to uses
    }
    public static int matchPlayersAndTrainers(int[] players, int[] trainers) {
        int result = 0;
        int minVal = Math.min(players.length, trainers.length);
        Arrays.sort(trainers);
        Arrays.sort(players);
        int j = trainers.length -1;
        int i = minVal - 1;
        while(i>=0) {
            if(players[i] <= trainers[j]){
                result++;
                j--;
                i--;
            }
            else{
                i--;
            }
            
            
        }
        return result;
    }
}

File: Medium/MaximumMatchingofPlayersWithTrainers.py
#  https://leetcode.com/problems/maximum-matching-of-players-with-trainers/description/

class Solution(object):
    def matchPlayersAndTrainers(self, players, trainers):
        players = sorted(players)
        trainers = sorted(trainers)
        minVal = min(len(players), len(trainers))
        j = len(trainers) - 1
        i = minVal - 1
        result = 0
        # print('Trainers: ', trainers)
        # print('Players: ', players)
        while (i >= 0):
            # print('i: ', i)
            # print('j: ', j)
            # print('player[i]: ', players[i])
            # print('trainers[j]: ', trainers[j])
            if (players[i] <= trainers[j]):
                result += 1
                j -= 1
                i -= 1
            else:
                i -= 1
            # print('result: ', result)
        return result


a = [2, 1]
b = [2, 1, 2, 2, 3, 3, 2, 4, 1, 1, 4, 1, 3, 3, 4, 1, 3, 2, 3, 2, 2, 3, 1, 2, 4]
obj = Solution()
print(obj.matchPlayersAndTrainers(a, b))
